File: sdks/java/examples/RegisterAgent.java
import dev.atap.sdk.ATAPClient;
import dev.atap.sdk.crypto.Ed25519KeyPair;
import dev.atap.sdk.model.Entity;

/**
 * Example: Register an agent entity with the ATAP platform.
 */
public class RegisterAgent {

    public static void main(String[] args) {
        // 1. Generate a new Ed25519 keypair
        Ed25519KeyPair keyPair = Ed25519KeyPair.generate();
        System.out.println("Generated public key: " + keyPair.getPublicKeyBase64());

        // 2. Register the agent (server will assign a DID)
        try (ATAPClient client = ATAPClient.builder()
                .baseUrl("http://localhost:8080")
                .build()) {

            Entity entity = client.entities.register("agent", "My AI Agent",
                    keyPair.getPublicKeyBase64(), null);

            System.out.println("Registered entity:");
            System.out.println("  ID:            " + entity.getId());
            System.out.println("  DID:           " + entity.getDid());
            System.out.println("  Client Secret: " + entity.getClientSecret());

            // 3. Now create an authenticated client using the returned credentials
            ATAPClient authedClient = ATAPClient.builder()
                    .baseUrl("http://localhost:8080")
                    .did(entity.getDid())
                    .keyPair(keyPair)
                    .clientSecret(entity.getClientSecret())
                    .build();

            // 4. Use the authenticated client
            Entity self = authedClient.entities.get(entity.getId());
            System.out.println("  Trust Level:   " + self.getTrustLevel());

            authedClient.close();
        }
    }
}

File: sdks/php/examples/register_agent.php
<?php

/**
 * Example: Register an agent entity and use the SDK.
 */

require_once __DIR__ . '/../vendor/autoload.php';

use Atap\Sdk\ATAPClient;

// 1. Register a new agent (no auth needed for registration)
$client = new ATAPClient([
    'base_url' => 'http://localhost:8080',
]);

$entity = $client->entities()->register('agent', 'My AI Agent');

echo "Registered agent:\n";
echo "  ID:            {$entity->id}\n";
echo "  DID:           {$entity->did}\n";
echo "  Client Secret: {$entity->clientSecret}\n";
echo "  Key ID:        {$entity->keyId}\n";

// 2. Create an authenticated client using the returned credentials
$authedClient = new ATAPClient([
    'base_url' => 'http://localhost:8080',
    'did' => $entity->did,
    'private_key' => $entity->privateKey, // Only if server generated the key
    'client_secret' => $entity->clientSecret,
]);

// 3. Check server health
$health = $authedClient->discovery()->health();
echo "\nServer health: {$health['status']}\n";

// 4. Look up the entity we just created
$fetched = $authedClient->entities()->get($entity->id);
echo "Fetched entity: {$fetched->name} (type: {$fetched->type})\n";
